c10/Stream.h

```diff
@@ -79,6 +111,15 @@ C10_API std::ostream& operator<<(std::ostream& stream, const Stream& s);
 
 } // namespace c10
 
+namespace std {
+  template <>
+  struct hash<c10::Stream> {
+    size_t operator()(c10::Stream s) const {
```

Comment:
nit: can this be noexcept? Depends on `std::hash<uint64_t>`.
In any case, this has to match the `noexcept` specifier of your other `std::hash` specialization because that one calls this one and is `noexcept`.